### PR TITLE
Debug improvements + rate limiting

### PR DESCRIPTION
**Pull request by inverse**
- Document other environment variables
- Add additional debug output
- Handle non 200 from dockerhub API + respect rate limiting

### CODE REVIEW

**Comment by inverse** [comment]
@ctbur what's left for this one?

File: mirror.go [hunk 1]
```diff
@@ -296,8 +296,8 @@ func (m *mirror) getRemoteTags() ([]RepositoryTag, error) {
                 log.Warningf("Failed to get %s, retrying", url)
                 retries -= 1
             } else if r.StatusCode == 429 {
-                log.Infof("Rate limited, sleeping")
-                time.Sleep(2 * time.Second)
+                log.Infof("Rate limited on %s, sleeping for 10s", url)
+                time.Sleep(10 * time.Second)
```
Comment (inverse) [inline mirror.go:300]: Should we make this come from an environment variable with 10 as the default?

Comment (ctbur) [reply mirror.go:300]: We can make it more sophisticated, but let's wrap up here and put the rest into an operations ticket.

**Comment by ctbur** [comment]
It seems to handle the occasional rate limit response better now, but it's still not correct. For example, there is this line in the logs:
```
time="2021-02-09T08:03:10Z" level=info msg="Repository mirror completed" full_repo=python3 num_tags=0 repo=python3
```
The number of tags is definitely not 0 for this repository.

File: mirror.go (hunk 1 above)
Comment (inverse) [reply mirror.go:300]: @ctbur anything else for this one?

**Review by ctbur — APPROVED** [review]

File: mirror.go (hunk 1 above)
Comment (inverse) [reply mirror.go:300]:
```
X-RateLimit-Limit: 180
X-RateLimit-Reset: 1614003678
X-RateLimit-Remaining: 178
```


These are the headers we get back - we could be smart and sleep

**Pull request merged**